plugins/scaffolder-backend/src/scaffolder/tasks/DatabaseTaskStore.ts

```diff
@@ -288,17 +303,22 @@ export class DatabaseTaskStore implements TaskStore {
         }
       };
 
-      const secrets = this.parseTaskSecrets(task);
-      return {
+      const ret: SerializedTask = {
         id: task.id,
         spec,
         status: 'processing',
         lastHeartbeatAt: task.last_heartbeat_at,
         createdAt: task.created_at,
         createdBy: task.created_by ?? undefined,
-        secrets,
-        state: getState(),
       };
+
+      this.events?.publish({
+        topic: 'scaffolder.task',
+        eventPayload: ret,
+      });
+
+      const secrets = this.parseTaskSecrets(task);
+      return { ...ret, secrets, state: getState() };
```

Comment:
Hey, I wonder what's the reason to exclude "state" from the event?

Reply:
Good point, I can add it but secrets should not go through.

Reply:
Secrets for sure not 😅 

"State" contains the checkpoint result, what means that if to re-run task again, instead of running already executed actions, it will take the outcome of these actions. Like for example, creation of GitHub repository, there can be stored the url to already created repo.